// https://github.com/vmagamedov/grpclib/blob/master/tests/dummy.proto
syntax = "proto3";

package dummy;

import "mypy_protobuf/extensions.proto";

message DummyRequest {
  string value = 1;
}

message DummyReply {
  string value = 1;
}

service DummyService {
  rpc UnaryUnary (DummyRequest) returns (DummyReply) {}
  rpc UnaryStream (DummyRequest) returns (stream DummyReply) {}
  rpc StreamUnary (stream DummyRequest) returns (DummyReply) {}
  rpc StreamStream (stream DummyRequest) returns (stream DummyReply) {}
}

service DummyAsyncService {
  option (mypy_protobuf.async_service) = true;
  rpc UnaryUnary (DummyRequest) returns (DummyReply) {}
}
